into/backends/sparksql.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import division, print_function, absolute_import
+
+from collections import Iterator
+import pandas as pd
+import itertools
+import datashape
+from datashape import dshape, Record, DataShape, Option, Tuple
+from datashape.predicates import isdimension
+from toolz import valmap
+
+from ..convert import convert
+from ..append import append
+from ..core import discover
+from ..into import convert, append
+from .spark import RDD, SchemaRDD, Dummy
+
+
+try:
+    from pyspark.sql import SQLContext
+    from pyspark.sql import (ByteType, ShortType, IntegerType, LongType,
+                             FloatType, DoubleType, StringType, BinaryType,
+                             BooleanType, TimestampType, DateType, ArrayType,
+                             StructType, StructField, MapType)
+except ImportError:
+    SchemaRDD = SQLContext = Dummy
+    ByteType = ShortType = IntegerType = LongType = FloatType = Dummy
+    MapType = DoubleType = StringType = BinaryType = BooleanType = Dummy
+    TimestampType = DateType = StructType = ArrayType = StructField = Dummy
+
+
+@convert.register(pd.DataFrame, SchemaRDD)
+def sparksql_dataframe_to_pandas_dataframe(rdd, dshape=None, **kwargs):
+    return pd.DataFrame(convert(list, rdd, **kwargs),
+                        columns=dshape.measure.names)
```

Comment:
This should probably just be `convert.register(list, SchemaRDD)` and then trust the network to move the data around to a `DataFrame`.  I hope that `DataFrame <- list` already does your columns trick.